app/helpers/progress_helper.rb

```diff
@@ -1,18 +1,19 @@
 module ProgressHelper
   def lesson_practice_key_for(stats)
-    if stats.try(:started?)
-      :continue_lesson
-    else
-      :start_lesson
-    end
+    key_or_default stats, :started?, :continue_lesson, :start_lesson
   end
 
-  #FIXME refactorme: similar methods
   def book_practice_key_for(student)
-    if student.try(:last_exercise)
-      :continue_practicing
+    key_or_default student, :last_exercise, :continue_practicing, :start_practicing
```

Comment:
I think this is better than what we had, but I still don't like having 4 parameters.  

However, I think that the whole navigation code should be refactored, since it is buggy and not easy to modify. See: 

* https://github.com/mumuki/mumuki-laboratory/issues/1105
* https://github.com/mumuki/mumuki-laboratory/issues/855